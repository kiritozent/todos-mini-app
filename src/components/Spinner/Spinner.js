import React from 'react';
import PropTypes from 'prop-types';
import { LoadingOutline } from '@ant-design/icons';
import { Spin } from 'antd';

function Spinner(props) {
  const { size, color } = props;
  let sizeNumber;
  if (typeof size === 'number') {
    sizeNumber = size;
  } else {
    switch (size) {
      case 'small':
        sizeNumber = 16;
        break;
      case 'medium':
        sizeNumber = 24;
        break;
      case 'large':
        sizeNumber = 32;
        break;
      default:
        break;
    }
  }
  return (
    <Spin
      indicator={
        <LoadingOutline style={{ fontSize: sizeNumber, color }} spin />
      }
    />
  );
}

Spinner.propTypes = {
  size: PropTypes.oneOfType([
    PropTypes.number,
    PropTypes.oneOf(['small', 'medium', 'large']),
  ]),
  color: PropTypes.string,
};

Spinner.defaultProps = {
  size: 24,
  color: '#1890ff',
};

export default Spinner;
