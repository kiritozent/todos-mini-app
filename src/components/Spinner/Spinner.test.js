import React from 'react';
import { shallow } from 'enzyme';
import Spinner from './Spinner';

describe('Spinner', () => {
  it('render without crashing passed', () => {
    const testCase = shallow(<Spinner />);
    expect(testCase).toMatchSnapshot();
  });
});
