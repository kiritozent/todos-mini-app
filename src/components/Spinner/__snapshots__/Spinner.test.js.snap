// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner render without crashing passed 1`] = `
<Spin
  indicator={
    <ForwardRef(LoadingOutlined)
      spin={true}
      style={
        Object {
          "color": "#1890ff",
          "fontSize": 24,
        }
      }
    />
  }
  size="default"
  spinning={true}
  wrapperClassName=""
/>
`;
